src/Diagnostics.ModelsAndUtils/Models/ResponseExtensions/Form.cs

```diff
@@ -0,0 +1,271 @@
+        {
+            // Returns true if we are able to add the input id to hashset
+            if(CurrentInputIds.Add(input.InputId))
+            {
+                input.CombinedId = input.InputId.ToString() + "." + FormId.ToString();
+                FormInputs.Add(input);             
+            }
+            else
+            {
+                throw new Exception($"Input ID {input.InputId} already exists for Form {FormId}. Please give a unique ID for the input");
+            }          
+        }
+
+        /// <summary>
+        /// Adds a list of form inputs to the form
+        /// </summary>
+        /// <param name="inputs">List of form inputs</param>
+        public void AddFormInputs(List<FormInput> inputs)
+        {
+            inputs.ForEach(input =>
+            {
+                AddFormInput(input);
+            });
+        }
+    }
+
+    public abstract class FormInput
+    {
+        /// <summary>
+        /// Represents Input ID
+        /// </summary>
+        public int InputId;
+
+        /// <summary>
+        /// Represents the Input Type
+        /// </summary>
+        public InputTypes InputType;
+
+        /// <summary>
+        /// Represents whether this is a required input
+        /// </summary>
+        public bool IsRequired;
+
+        /// <summary>
+        /// Represents the label of the input 
+        /// </summary>
+        public string Label;
+
+        /// <summary>
+        /// Combines the parent form id and input id
+        /// </summary>
+        public string CombinedId;
+
+        /// <summary>
+        /// Creates an input with given id and input type
+        /// </summary>
+        /// <param name="id">unique id for the input</param>
+        /// <param name="inputType">Input type for the input</param>
+        /// <param name="label">The label of the input</param>
+        /// <param name="isRequired">Indicates whether this is a required input</param>
+        public FormInput(int id, InputTypes inputType, string label = "", bool isRequired = false)
+        {
+            InputId = id;
+            InputType = inputType;
+            Label = label;
+            IsRequired = isRequired;
+        }
+    }
+
+    /// <summary>
+    /// Textbox class used to create a textbox input
+    /// </summary>
+    /// <example>
+    /// This sample shows how to create a textbox input
+    /// <code>
+    /// Textbox input1 = new Textbox(1, "Role Instance name", true);
+    /// </code>
+    /// </example>
+    public class Textbox: FormInput
+    {
+        /// <summary>
+        /// Creates an instance of Textbox class
+        /// </summary>
+        /// <param name="id">Unique id for the input</param>
+        /// <param name="label">Label for the textbox</param>
+        /// <param name="isRequired">Indicates if it is a required input</param>
+        public Textbox(int id, string label = "", bool isRequired = false): base(id, InputTypes.TextBox, label, isRequired)
```

Comment:
isnt Label mandatory instead of optional?

Reply:
Yes true. Will change it.